src/quo2/components/dividers/strength_divider/view.cljs

```diff
@@ -0,0 +1,75 @@
+(ns quo2.components.dividers.strength-divider.view
+  (:require [react-native.core :as rn]
+            [quo2.components.dividers.strength-divider.style :as style]
+            [quo2.components.markdown.text :as text]
+            [quo2.foundations.colors :as colors]
+            [react-native.linear-gradient :as linear-gradient]
+            [react-native.svg :as svg]
+            [utils.i18n :as i18n]
+            [quo2.components.icon :as icon]))
+
+(def status-list
+  {:very-weak   {:default-text (i18n/label :t/strength-divider-status-very-weak)
+                 :color        colors/danger-60
+                 :percentage   20}
+   :weak        {:default-text (i18n/label :t/strength-divider-status-weak)
+                 :color        (colors/custom-color :orange 60)
+                 :percentage   40}
+   :okay        {:default-text (i18n/label :t/strength-divider-status-okay)
+                 :color        (colors/custom-color :yellow 60)
+                 :percentage   60}
+   :strong      {:default-text (i18n/label :t/strength-divider-status-strong)
+                 :color        colors/success-60
+                 :percentage   80}
+   :very-strong {:default-text (i18n/label :t/strength-divider-status-very-strong)
+                 :color        colors/success-60
+                 :percentage   100}
+   :info        {:color colors/white-opa-40}
+   :alert       {:color colors/danger-60}})
+
+(def status-indicator-radius 6.5)
+
+(defn view
+  "Options
+   - `status` Status keyword (`:very-weak`/`:weak`/`:okay`/`:strong`/`:very-strong`/`:info`/`:alert`)
+
+   `text` message string(only works when status is `info`/`alert`)"
+  [{:keys [status] :or {status :very-weak}} text]
+  (let [{:keys [color percentage default-text]} (status-list status)
+        status-circumference                    (* 2 Math/PI status-indicator-radius)]
+    [linear-gradient/linear-gradient
```

Comment:
The component looks great! :)  congrats!

I was thinking about a small refactoring, you could apply my suggestions if you think the approach is better.

I noticed the `linear-gradient` component is not necessary when we want to render the `:alert` or `:info`  statuses, additionally, I'd suggest to create smaller funcitons that wrap a single "behaviour" or, in this case, a single "sub component", because it makes the reading and future modifications easier.

I refactored a little the code and looks as follows, also added some suggestions and comments about the refactor:

1. Create a `circle-percentage` (or a better name) function that only renders the circle
```clojure
(defn circle-percentage [{:keys [color percentage]}]
  (let [status-indicator-radius 6.5 ; <- instead a global var, it clould be a local binding
        status-circumference    (* 2 Math/PI status-indicator-radius)]
    ;; linear-gradient is only used here, instead of wrapping the whole component
    [linear-gradient/linear-gradient {:colors [(colors/alpha color 0.05) (colors/alpha color 0)]}
     [svg/svg
      ;; Rest of your svg code
      ]]))
```

2. Create a function to draw the visual indicator:
```clojure
(defn indicator [status] ; <- Probably not the best name, could be enhanced, the idea is
                         ; to be clear about it is a graphic representation of the status
  (let [{:keys [color percentage]} (status-list status)]
    (case status
      :info  nil ; <- info doesn't render any visual indicator
      :alert [icon/icon :i/alert {:color color :size 16}]
      [circle-percentage {:color      color
                          :percentage percentage}])))
```

and finally, create your `view` function joining the parts:
```clojure
(defn view
  "docstring"
  [{:keys [status] :or {status :very-weak}} text]
  (let [{:keys [color default-text]} (status-list status)] ; <- I think is better to change the `status-list` name
                                                           ; because the `list` suffix could indicate it is a list or vector
                                                           ; which is not the case.
    [rn/view {:style container, :accessibility-label :strength-divider} ; <- only this view wraps the component
     [indicator status] ; <- it's clear we have an indicator
     [text/text {:size   :paragraph-2 ; <- It's clear we have a text
                 :weight :medium
                 :style  (style-text color)}
      ;;The previuous `if` is more complex than needed, if we have a `default-text` in `status-list`,
      ;; then we ignore the custom given text.
      (or default-text text)]]))
```

Of course, you can modify anything about this suggestion. I tried it and it works. What do you think about it?

Reply:
Great! Thanks for taking the time to give such a detailed review. I will definitely update the implementation based on your suggestions.

Reply:
From my understanding, all the statuses including `: info` & `: alert` has a background linear gradient. So I think the gradient is common for all components. Could you please check and verify? 

- Component Design: https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?node-id=14826%3A196114&t=JLWf3YPHtg24GrUI-0
- Screen using alert: https://www.figma.com/file/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=2265%3A254313&t=3io35HqiowwAUulo-4
- Screen using info: https://www.figma.com/file/o4qG1bnFyuyFOvHQVGgeFY/Onboarding-for-Mobile?node-id=2241%3A251839&t=3io35HqiowwAUulo-4

Reply:
@ajayesivan I'm sorry I was a little bussy, I'm looking at this rn

Reply:
Yes! you're totally right! sorry for that.
Then the linear-gradient must wrap the whole component! please exclude that suggestion but keep in mind the other ones :)

Reply:
@ulisesmac I have updated the pr based on your suggestions, please have a look.

Thanks again for the detailed review.